R/missing_argument_linter.R

```diff
@@ -4,6 +4,35 @@
 #'
 #' @param except a character vector of function names as exceptions.
 #' @param allow_trailing always allow trailing empty arguments?
+#'
+#' @examples
+#' # will produce lints
+#' lint(
+#'   text = "mean(x, )",
+#'   linters = missing_argument_linter()
+#' )
+#'
+#' # okay
+#' lint(
+#'   text = "mean(x, na.rm = FALSE)",
+#'   linters = missing_argument_linter()
+#' )
+#'
+#' lint(
+#'   text = "mean(x)",
+#'   linters = missing_argument_linter()
+#' )
+#'
+#' lint(
+#'   text = "mean(x, )",
+#'   linters = missing_argument_linter(except = "mean")
+#' )
+#'
+#' lint(
+#'   text = "mean(x, )",
+#'   linters = missing_argument_linter(allow_trailing = TRUE)
+#' )
```

Comment:
not a good example I think, since this may be erroneous to use. better to use a list2() example, maybe filter() or tibble() or mutate(), to demonstrate a case where the lint is more subjective as opposed to avoiding errors

Reply:
I have changed the example, but still wanted to understand your comment.

Under what circumstances will this be the case?

> this may be erroneous to use

``` r
x <- c(1, 2, 3)
mean(x, )
#> [1] 2
```

<sup>Created on 2022-10-10 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

Reply:
You're right, I was speaking a bit more generally (hadn't check if `mean(x, )` works)...  point being it's very hard to tell at a glance whether code like that will run or not (and hence a lint is really good), whereas I thing `allow_trailing=TRUE` is intended to allow a common source of false positives through (namely, `list2()` usages)

Reply:
Gotcha!